openshift-knative-operator/pkg/serving/extension.go

```diff
@@ -164,6 +164,11 @@ func (e *extension) Reconcile(ctx context.Context, comp base.KComponent) error {
 		common.ConfigureIfUnset(&ks.Spec.CommonSpec, monitoring.ObservabilityCMName, monitoring.ObservabilityBackendKey, "none")
 	}
 
+	// DomainMapping cleanup
+	if err := e.cleanupDomainMapping(ctx, ks); err != nil {
+		//TODO: we don't need to propagate the error further?
```

Comment:
I think we should either propagate it and so another reconciliation will happen or do block and retry until we delete the resources.